Internal config parse

gandarez opened the pull request:
This PR moves the responsability of loading `internal-config` file to the main entry likewise for reading the regular config file and also gives back the initial approach of reading any param using Viper. It was also breaking tests on my side because there was backoff set at my waka config file. In addition it also includes a fix for filtering heartbeat where the error message was truncated.

gandarez committed:
Fix error message at filtering

gandarez committed:
Fix internal config file parse

dron22 commented:
Hey @gandarez! I will review this later today at about 8pm CET  :+1:

dron22 reviewed (approved):
Nice one :+1: Great that we read everything via VIper now :rocket: 

Should also be reviewed by @alanhamlett, as I was not really involved, when we added this internal config file support.

alanhamlett commented:
LGTM... I manually parsed the internal ini file because I couldn't find a way to parse two config files with the same Viper instance.

alanhamlett commented:
Maybe out of scope in this PR, but can we add a test for corrupted internal config file and make sure the heartbeat is saved to offline db?

Something like:
```ini
[internal
backoff_at = abc
```

(missing closing square bracket)

gandarez commented:
I don't think it's out of scope. I'll add a test for malformed internal config file.

gandarez commented:
I've added two new integration tests where one for regular malformed config and another one for malformed internal config. I also fixed reading multiple configs using Viper by changing from `ReadInConfig` to `MergeInConfig` and added a test for that.
